From a982f487e575497244591b78abac270fafae1827 Mon Sep 17 00:00:00 2001
From: Eric Anholt <eric@anholt.net>
Date: Mon, 9 Nov 2009 14:57:34 -0800
Patch-mainline: 2.6.34
References: fate#310916
Git-commit: 21099537dbacc5c8999d833e6bfd1b72edd89189
Subject: [PATCH 0454/2588] drm/i915: Correct locking in the modesetting
 failure path, fixing a BUG_ON.

Signed-off-by: Eric Anholt <eric@anholt.net>
(cherry picked from commit 21099537dbacc5c8999d833e6bfd1b72edd89189)

Signed-off-by: Takashi Iwai <tiwai@suse.de>
---
 drivers/gpu/drm/i915/i915_dma.c |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/drivers/gpu/drm/i915/i915_dma.c b/drivers/gpu/drm/i915/i915_dma.c
index 76797e9..f1dbafa 100644
--- a/drivers/gpu/drm/i915/i915_dma.c
+++ b/drivers/gpu/drm/i915/i915_dma.c
@@ -1411,7 +1411,9 @@ static int i915_load_modeset_init(struct drm_device *dev,
 	return 0;
 
 destroy_ringbuffer:
+	mutex_lock(&dev->struct_mutex);
 	i915_gem_cleanup_ringbuffer(dev);
+	mutex_unlock(&dev->struct_mutex);
 out:
 	return ret;
 }
-- 
1.7.6

